### PR TITLE
Add skeleton and basic functionalities

### PR DESCRIPTION
**Pull request by longquanzheng**

### CODE REVIEW

File: iwf/workflow_state.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+from iwf.workflow_state_options import WorkflowStateOptions
+
+T = TypeVar("T")
+
+
+class WorkflowState(ABC, Generic[T]):
+    """WorkflowState is the interface to define a workflow state."""
+
+    @abstractmethod
+    def get_input_type(self) -> type[T]:
```
Comment (zhxnlai) [inline iwf/workflow_state.py:19]: this does need to be provided by a subclass? https://stackoverflow.com/questions/48572831/how-to-access-the-type-arguments-of-typing-generic

File: iwf/workflow_options.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+from iwf_api.models import IDReusePolicy, WorkflowRetryPolicy
+
+
+@dataclass
+class WorkflowOptions:
+    workflow_id_reuse_policy: IDReusePolicy
+    workflow_cron_schedule: str
+    workflow_retry_policy: WorkflowRetryPolicy
+    InitialSearchAttributes: dict[str, Any]
```
Comment (zhxnlai) [inline iwf/workflow_options.py:12]: should this be `initial_search_attributes`?

File: iwf/registry.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+from iwf.errors import WorkflowDefinitionError
+from iwf.state_def import StateDef
+from iwf.workflow import ObjectWorkflow, get_workflow_type
+from iwf.workflow_state import get_state_id
+
+
+class Registry:
+    def __init__(self):
+        self._workflowStore = {}
+        self._startingStateStore = {}
```
Comment (zhxnlai) [inline iwf/registry.py:10]: nit: should this be snakecase?

File: iwf/workflow_state.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                    BackoffCoefficient: 2
+        Returns: WorkflowStateOptions
+        """
+        pass
+
+
+def get_state_id(state: WorkflowState) -> str:
+    options = state.get_state_options()
+    if options is None or options.state_id is None:
+        return state.__class__.__name__
```
Comment (zhxnlai) [inline iwf/workflow_state.py:100]: should this be a fully qualified name? https://gist.github.com/clbarnes/edd28ea32010eb159b34b075687bb49e

File: iwf/workflow_state.py (hunk 1 above)
Comment (longquanzheng) [reply iwf/workflow_state.py:19]: type is "needed" but T is not. Type will be needed for deserailize the payload into the state input. I can be None if no input. Or some primitive types

Comment (longquanzheng) [reply iwf/workflow_state.py]: Great idea! Removed this method:D

File: iwf/tests/worker_server.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+def handle_execute():
+    req = WorkflowStateExecuteRequest.from_dict(request.json)
+    resp = _worker_service.handle_workflow_state_execute(req)
+    return resp.to_dict()
+
+
+@_flask_app.errorhandler(Exception)
+def internal_error(exception):
+    print("500 error caught")
+    print(traceback.format_exc())
```
Comment (zhxnlai) [inline iwf/tests/worker_server.py:43]: Consider using the standard logging interface: https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library

**Pull request merged**